### PR TITLE
Update of package versions.

### PR DESCRIPTION
**Pull request by 0xorial**
Bumping versions of different dependencies. Still [looking for a better way](https://github.com/dsherret/ts-morph/issues/1341) to get string type

### CODE REVIEW

**Comment by rhys-vdw** [comment]
Nice work. Did you want to wait for the string thing before merging?

(can you remind me what the string type is for?)

**Comment by 0xorial** [comment]
Thanks!

> Did you want to wait for the string thing before merging?

It technically works, but that temporary code doesn't seem good. It is indeed the only thing I am waiting - don't know if it's better to merge and make another PR later, or wait for a better way for that...

This piece of code was checking types derived from `Record<string, ...>`. In the new version type alias parameters are simply 'T' and 'K', didn't find how to resolve the specific type. But found something that seems more universal to detect that type has an index `[k: string]: T`, so used it instead. Only problem then is that our function generating the condition for index signature takes type of the key and type of the value. It is quite big so I didn't want to rewrite it, but perhaps it is a better solution.

File: src/index.ts [hunk 1]
```diff
@@ -406,7 +406,24 @@ To disable this warning, put comment "${suppressComment}" before the declaration
       conditions.push(
         indexSignaturesCondition(
           varName,
-          indexSignatures,
+          indexSignatures.map(x => {
+            const keyType = x.keyType.isString()
```
Comment (rhys-vdw) [inline src/index.ts:410]: Could we make a helper method like...

```js
function typeToString(Type keyType) {
  switch (true) {
    case type.isString(): return 'string';
    case type.isNumber(): return 'number';
    // ...
    default: throw new Error(...)
  }
}
```

File: tests/import.test.ts [hunk 2]
```diff
@@ -11,24 +12,26 @@ interface TestDefinition {
...
-  inputContents: string
-  expectedContents: string
-  message: string
-  constructor(message: string, inputFile: string, guardFile: string) {
-    this.inputContents = inputFile
-    this.expectedContents = guardFile
-    this.message = message
-  }
+  constructor(
+    public message: string,
```
Comment (rhys-vdw) [inline tests/import.test.ts:21]: This is cool! I haven't used TypeScript for ages so I never got to use this one. :-)

**Review by rhys-vdw — COMMENTED** [review]
Alright, this looks good to me (and I trust the tests). I've asked for a change, but I'm happy for you to make the call. I think you have merge rights, so go ahead and merge after the change.

Great work on some much-needed maintenance, super happy to see the project being supported. <3

**Comment by rhys-vdw** [comment]
Nice work. Releasing new version: https://github.com/rhys-vdw/ts-auto-guard/actions/runs/3293619012

**Pull request merged**